h/views/groups.py

```diff
@@ -112,18 +113,12 @@ def _update_group(self, appstruct):
 @view_config(route_name='group_read',
              request_method='GET',
              renderer='h:templates/groups/share.html.jinja2',
-             effective_principals=security.Authenticated)
+             effective_principals=security.Authenticated,
+             has_permission='read')
```

Comment:
We make it so that this is only called if you have read permission, so that JoinGroupController below will be called if you don't